forte/data/data_pack.py

```diff
@@ -1322,10 +1341,10 @@ def _generate_link_entry_data(
                 )
 
             a_dict["parent"].append(
-                np.where(data[parent_type]["tid"] == link.parent)[0][0]
+                np.where(data[parent_type]["tid"] == link.parent.tid)[0][0]  # type: ignore
```

Comment:
This is a hacky fix. But this stems from another problem that we are attempting solve through #835